### DISCUSSION TITLE
Faster level music simmilar to the genesis games

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#166] [Show and tell]
This is a tutorial for who ever wants to add faster level music for speed shoes 

1.add a export variable to the level.gd 
    for example

Export var music ==0

2. Next add a new themes array to the global and don't forget a variable 

      var fsthemes = [preload("res://Audio/Soundtrack/4-01.-Angel-Island-Zone-Act-1-_Fast_.ogg"),preload("res://Audio/Soundtrack/4-02.-Angel-Island-Zone-Act-2-_Fast_.ogg")]
var currentfsTheme = 0
var Musics =0

3. connect the global variable to the variable you made in level.gd
    
    Global.Musics =music

4. Now go to the monitor script and where it says 
add a call for the global variable that you set in global.gd

    if !playertouch.get("isSuper"): 

    It should look like this 

    if !playerTouch.get("isSuper") and Global.Musics ==0:
	playerTouch.shoeTime = 3
	playerTouch.switch_physics()
	Global.currentfsTheme =0
	Global.effectTheme.stream =Global.fsthemes[Global.currentfsTheme]
	Global.effectTheme.play() 

 And that's it you can now have speed shoes music similar to the classic sonic games

**Comment by DimensionWarped** [comment]
Isn't this going to result in the music resetting whenever you get the speed shoes upgrade/lose the speed shoes upgrade?

Part of what made the speed shoes upgrade effect interesting was that the music changed in-place... though this was easily doable only because the Genesis generates its music audio via a system similar to MIDI audio rather than playing back recordings.

There is a midi player (one that even supports sound fonts) for Godot that I saw a while back that looked pretty cool for people interested in that sort of thing. An alternative approach would be to figure out how to adjust your playback time when switching songs based on percentage of progress from the old song. Might want to set up some kind of cross-fade and/or drum fill between the two to mask the transition too.

**Reply by Raptorq1** [reply]
I was trying to make it work like how it worked in sonic 3 air

**Reply by DimensionWarped** [reply]
I wanted to check, but I can't verify because it's damn near impossible to actually find speed shoes in Sonic 3.

**Reply by Raptorq1** [reply]
This how it works in  sonic 3 air 

[untitled.txt](https://github.com/user-attachments/files/16090759/untitled.txt)

**Reply by DimensionWarped** [reply]
Looks to me like it speeds up the audio in place rather than switching which song is played.

**Reply by Raptorq1** [reply]
Okay then how would I set it up

**Reply by DimensionWarped** [reply]
Well I explained that in my first response... your options are to use something more like a midi player and totally change up which music you are using, have two versions of each song (one already sped up) like you are doing above and figure out how to advance the playback of the songs as needed to match with the other song while also masking the seam somehow (tricky!) or else figure out how to use the audio bus to make something 'good enough'. There are actually more ways to skin this cat than I can name. If you want an example of one that would require loads of technical skill you could port a Yamaha YM3438 emulator to Godot and use that to make/play your music.

For you, the approach you are using is probably fine, you don't need to be accurate to the games.

**Reply by Raptorq1** [reply]
This is how Christian whitehead did it for sonic 1 and sonic 2 mobile
